Katello/katello
# Fixes #37517 - Upgrade theforeman-rubocop gem to the  v0.1.0

#### What are the changes introduced in this pull request?

- Upgrade theforeman-rubocop gem to version 0.1.0, https://community.theforeman.org/t/standardizing-rubocop-with-theforeman-rubocop/37239/1#use-theforeman-rubocop-010-in-foreman-and-all-plugins-1
- Changed the RuboCop configuration to inherit from the `lenient.yml` style provided by `theforeman-rubocop`, https://github.com/theforeman/theforeman-rubocop?tab=readme-ov-file#not-intrusive-style---lenient
- Removed `Rails` and `Style/Documentation` cops from the configuration as they are already included in the `lenient.yml` style.

#### Considerations taken when implementing this change?

- Chose `lenient.yml` because it aligns well with the existing RuboCop style followed by this repo.
- Removed `Rails` and `Style/Documentation` cops to avoid redundancy and conflicts since they are already covered by `lenient.yml`.

#### What are the testing steps for this pull request?

1. Run `bundle install` to ensure all dependencies are correctly installed, including the updated `theforeman-rubocop` gem.
2. Execute `bundle exec rubocop` to verify that the new RuboCop configuration is being applied without errors.
3. Review the output to ensure there are no unexpected offenses or configuration issues.
